docs/key_concepts.rst

```diff
@@ -0,0 +1,87 @@
+Key Concepts
+============
+
+This section introduces nameko's central concepts.
+
+Anatomy of a Service
+--------------------
+
+A nameko service is just a Python class. The class encapsulates the application logic in its methods and declares any :ref:`dependencies <dependencies>` as attributes.
+
+Methods are exposed to the outside world with :ref:`entrypoint <entrypoints>` decorators.
+
+.. literalinclude:: examples/example.py
+
+.. _entrypoints:
+
+Entrypoints
+^^^^^^^^^^^
+
+Entrypoints are gateways into the service methods they decorate. While a service is running, an entrypoint may "fire" and the decorated method would be executed by a service :ref:`worker <workers>`.
```

Comment:
Bit opaque at this point in the docs. Worth mentioning what an entrypoint normally is i.e. something responding to an RPC call or handling an event.